default/python/universe_generation/empires.py

```diff
@@ -154,11 +169,15 @@ def find_home_systems_for_min_jump_distance(self, systems_pool, min_jumps):
                 # remove all neighbors from the local pool
                 local_pool -= set(get_systems_within_jumps(member, min_jumps))
 
-            # Count complete misses when number of candidates is not close to the target.
-            if len(candidate) < (self.num_home_systems - 4):
-                num_complete_misses_remaining -= 1
-
             if len(candidate) < self.num_home_systems:
+
+                # Break if the normal distribution of failed candidates is too much smaller than
+                # the target number of systems
+                lens_of_failed_candidates.append(len(candidate))
+                mean_len_of_failed_candidates = sum(lens_of_failed_candidates) / float(len(lens_of_failed_candidates))
+                if mean_len_of_failed_candidates < lowest_continuable_mean_len_of_candidates:
+                    break
```

Comment:
Ok, I'm officially out-mathed. 😉 

From the little bit I understand of the above exchange, I want to suggest this:

When I suggested to scale the fixed variance of 4, I have been thinking of something simple like `num_home_systems/10 + 1`. Judging by the numbers of the example above (999 system map with 39 home systems), a fixed variance of 4 would probably be sufficient for maps with ten thousands of systems and 100+ home systems, a limit we'll probably never hit, at least not within the next 10-20 years.

Which means, returning to the prior version, where we had a check for a "complete miss" (a complete miss being `len(candidate) < num_home_systems - 4`) and exit early after 5 such "complete misses" sounds like a reasonable approach to me.

This should be negligible performance-wise, while not as aggressive as quitting immediately when a candidate set doesn't yield at least `num_home_systems` systems. The cases of that happening because of a very unlucky random selection might be very small, but I still want that cases be avoided as best as possible.

> The total improvement is an increase in s of 1. This is a lot of angst for minimal gain

I beg to differ. Even when considering the difference between a minimum spacing of 10 (which is the highest minimum spacing we begin with) vs. 9, I wouldn't call that "minimal". And it's only getting more significant from there on.

And the "lot of angst" is deliberate. Getting the required number of home systems as far away from each other as possible, and if possible from the strictest pool has absolute priority over speed.

The latter condition is so important because using the more relaxed pools means the min system and planets within a certain vicinity limit most likely won't be met by all home systems, which means planets have to be added to the systems in the vicinity of those home systems, which I want to avoid as much as possible.